pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java

```diff
@@ -514,8 +514,9 @@ void run() throws PulsarAdminException {
         @Override
         void run() throws Exception {
             String topic = validateTopicName(params);
-            Map<String, String> map = new HashMap<>();
-            if (metadata != null) {
+            Map<String, String> map = null;
+            if (metadata != null && !metadata.isEmpty()) {
+                map = new HashMap<>();
```

Comment:
Are the changes necessary? It looks like `map`, as the 3rd argument of `Topics#createPartitionedTopic`, is now `null` rather than an empty `HashMap` if `metadata` is empty.

Reply:
That is intentional